DA/Internal/Desugar.hs

```diff
@@ -184,9 +184,11 @@ class HasExerciseGuarded t c r | t c -> r where
 _exerciseDefault : HasExerciseGuarded t c r => ContractId t -> c -> Update r
 _exerciseDefault = exerciseGuarded (const True)
 
-_exerciseInterfaceGuard : forall i t. HasFromInterface t i => (t -> Bool) -> (i -> Bool)
-_exerciseInterfaceGuard pred iface =
-  optional False pred (fromInterface iface)
+_exerciseInterfaceGuard : forall i t.
```

Comment:
As discussed offline: 
Changed to match https://github.com/digital-asset/daml/blob/69296455f3acc4b97833a800dc285af34ecf7b67/compiler/damlc/daml-stdlib-src/DA/Internal/Interface.daml#L157